public/pages/AnomalyCharts/containers/FeatureBreakDown.tsx

```diff
@@ -45,9 +48,68 @@ interface FeatureBreakDownProps {
   showFeatureMissingDataPointAnnotation?: boolean;
   rawAnomalyResults?: Anomalies;
   isFeatureDataMissing?: boolean;
+  isHCDetector?: boolean;
+  selectedHeatmapCell?: HeatmapCell;
 }
 
 export const FeatureBreakDown = React.memo((props: FeatureBreakDownProps) => {
+  const getFeatureDataForChart = (
+    anomaliesResult: Anomalies,
+    featureId: string
+  ) => {
+    const originalFeatureData = get(
+      anomaliesResult,
+      `featureData.${featureId}`,
+      []
+    );
+    if (props.isHCDetector) {
+      if (props.selectedHeatmapCell) {
+        const anomaliesFound = get(anomaliesResult, 'anomalies', []);
+        const filteredFeatureData = [];
+        for (let i = 0; i < anomaliesFound.length; i++) {
+          const currentAnomalyData = anomaliesResult.anomalies[i];
+          if (
+            !isEmpty(get(currentAnomalyData, 'entity', [] as EntityData[])) &&
+            get(currentAnomalyData, 'entity', [] as EntityData[])[0].value ===
+              props.selectedHeatmapCell.entityValue &&
+            get(currentAnomalyData, 'plotTime', 0) >=
+              props.selectedHeatmapCell.dateRange.startDate &&
+            get(currentAnomalyData, 'plotTime', 0) <=
+              props.selectedHeatmapCell.dateRange.endDate
+          ) {
+            filteredFeatureData.push(originalFeatureData[i]);
+          }
+        }
+        return filteredFeatureData;
+      } else {
+        return [];
+      }
+    } else {
+      return originalFeatureData;
+    }
+  };
+  const getAnnotationData = () => {
+    if (props.isHCDetector) {
+      if (props.selectedHeatmapCell) {
+        return filterWithHeatmapFilter(
+          props.annotations,
+          props.selectedHeatmapCell,
+          true,
+          'coordinates.x0'
+        );
+      } else {
+        return [];
+      }
+    } else {
+      return props.annotations;
+    }
+  };
+  const getDateRange = () => {
+    if (props.isHCDetector && props.selectedHeatmapCell) {
+      return props.selectedHeatmapCell.dateRange;
+    }
+    return props.dateRange;
+  };
```

Comment:
Duplicate function in `AnomalyOccurrenceChart`. Maybe refactor this function and the other `get*` functions in both `AnomalyOccurrenceChart` and `FeatureBreakDown` into a common helpers file?

Reply:
sure. will add it.